c7n/filters/metrics.py

```diff
@@ -128,11 +128,14 @@ def process(self, resources, event=None):
         duration = timedelta(days)
 
         self.metric = self.data['name']
-        self.end = datetime.utcnow()
 
-        # Adjust the start time to gracefully handle CloudWatch's retention schedule, which rolls up
-        # data points progressively (1 minute --> 5 minutes --> 1 hour) over time.
-        self.start = (self.end - duration).replace(minute=0)
+        # If the policy does not specify a period, derive a single period using
+        # the number of days. Ensure that the period is a clean multiple of
+        # hours so we fetch data points with the longest retention.
+        #
+        # Reference: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Metric  # noqa
+        self.end = datetime.utcnow().replace(minute=0, second=0, microsecond=0)
```

Comment:
some folks want intra hour periods, like last 15m etc.. just making sure we're compatible, the comment seems like it would but the diff is elided if that logic is happening elsewhere.

Reply:
Dang, no this would break that expectation in service of making longer periods work better.

The most technically accurate thing I could think to do would be consider how far back the filter needs to look, and nudge the times to they line up with an even 1 min / 5 min / 1 hour division to balance resolution and retention (using the retention schedule from [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Metric)). That seemed overly complicated but maybe it's necessary.

Reply:
Any chance this "fix" is going to make it into an upcoming release? The fix as it stands work for our needs but I understand it might not work for those wanting to check shorter periods (i.e. less than an hour)?

Reply:
Context-switching back to this I tweaked and tested the logic so hopefully the motivation and behavior around determining start/end times is a bit clearer. These changes should help fix policies looking at long periods, without breaking the ones looking at short periods. Any checks on your end @stepkirk would be appreciated since I know you have a vested interest in seeing this live and experience with past breakage :)